numpyro/infer_util.py

```diff
@@ -12,17 +12,29 @@ def log_density(model, model_args, model_kwargs, params):
     :param dict model_kwargs`: kwargs provided to the model.
     :param dict params: dictionary of current parameter values keyed by site
         name.
+    :param bool skip_transforms: whether to compute log probability of a site
+        (if its prior is a transformed distribution) in its base distribution
+        domain.
     :return: log of joint density and a corresponding model trace
     """
-    model = substitute(model, params)
+    if skip_dist_transforms:
+        model = substitute(model, base_param_map=params)
+    else:
+        model = substitute(model, params)
     model_trace = trace(model).get_trace(*model_args, **model_kwargs)
     log_joint = 0.
     for site in model_trace.values():
         if site['type'] == 'sample':
             value = site['value']
             intermediates = site['intermediates']
-            log_prob = np.sum(site['fn'].log_prob(value, intermediates) if intermediates
-                              else site['fn'].log_prob(value))
+            if intermediates:
+                if skip_dist_transforms:
+                    log_prob = site['fn'].base_dist.log_prob(intermediates[0][0])
```

Comment:
This is assuming a certain form for the intermediates, namely, `(val, log_abs_det_jacobian)`. Are we assuming that this is the interface that all `TransformedDistribution` classes will follow?

Reply:
I think subclasses of `TransformedDistribution` should not redefine `transform_with_intermediates` method. Then the pattern `[[val1, transform_inter1], [val2, transform_inter2], ...]` will be consistent.

Reply:
Hmm..how about checking for `base_value` instead? Then you should be able to use that directly here. `site['fn'].base_dist.log_prob(base_value)`.

Reply:
understood! I will make a change